### ISSUE TITLE
Use rustup.sh for `language: rust`

### ISSUE THREAD

**Issue by SimonSapin** [travis-ci/travis-ci#3704]
Currently, one can use:

```yaml
language: rust
rust:
    - nightly
    - 1.0.0-beta
```

in `.travis.yml`. The version numbers are used directly to [get a tarball URL](https://github.com/travis-ci/travis-build/blob/de292bd16d964d2223010a61d912118479cfc014/lib/travis/build/script/rust.rb#L5-L8).

Unfortunately, aside from `nightly`, this scheme only supports installing precise Rust versions.

Rather than downloading tarballs directly, travis-build could be changed to use [`rustup.sh`](https://github.com/rust-lang/rustup), a script that knows about Rust release channels and can install various Rust versions:

> Install nightly.
> 
> ```
> rustup.sh --channel=nightly
> ```
> 
> Install nightly archives.
> 
> ```
> rustup.sh --channel=nightly --date=2015-04-09
> ```
> 
> Install explicit versions.
> 
> ```
> rustup.sh --revision=1.0.0-beta
> ```

This could enable usage such as:

```yaml
language: rust
rust:
    - nightly
    - beta
    - stable
    - 1.4
```

to test e.g. in whatever is the latest of each release channel, plus one specific (then old) version.

**Comment by mvdnes** [comment]
Resolved with travis-ci/travis-build#430 :star2:

**Comment by BanzaiMan** [comment]
Indeed.
